src/mlir/dialect/VeronaOps.cc

```diff
@@ -111,6 +133,32 @@ static LogicalResult verify(verona::ClassOp classOp)
   return success();
 }
 
+static LogicalResult verify(verona::WhileOp whileOp)
+{
+  auto& body = whileOp.body();
+
+  // While blocks must have a condition on the first block
+  // with a LoopExit operation.
```

Comment:
Similarily, this is quite a broad requirement.
What if the program looks like, using a ternary operator:
```
while (x ? y : z) { ... }
```

Then the IR will look like
```mlir
while {
  cond_br %x, ^bb1, ^bb2
^bb1:
  br ^bb3(%y)
^bb2:
  br ^bb3(%z)
^bb3(%0):
  loop_exit %0
}
```

Then the first basic block does not contain a `loop_exit`.

(We might not have this form of ternary operator, but I'm sure we can have expression that generate more than one basic block)

You could, as a requirement, have that all paths through the loop must go through a `loop_exit`, but I'm not even sure I see the point of this requirement.

Reply:
`loop_exit` is not required to be in the first basic block, but it separates the basic blocks that compose the condition from those that compose the body. It's precisely because both condition and body could have multiple basic blocks that I want to split the region.

Reply:
Updated the check to get this one right, too.